### PR TITLE
chore: explicit behavior for nonce update persistence

### PR DESCRIPTION
**Pull request by nbaztec**
# What :computer: 
* Defines nonce update behavior for tx caller, explicitly.

### CODE REVIEW

File: crates/cheatcodes/src/inspector.rs [hunk 1]
```diff
@@ -1841,8 +1868,7 @@ where {
             accesses: self.accesses.as_mut(),
             persisted_factory_deps: Some(&mut self.persisted_factory_deps),
             paymaster_data: self.paymaster_params.take(),
-            should_update_nonce: self.broadcast.is_some() ||
-                self.zk_should_update_nonce.take().unwrap_or_default(),
+            persist_nonce_update: self.broadcast.is_some() || self.zk_persist_nonce_update.get(),
```
Comment (elfedy) [inline crates/cheatcodes/src/inspector.rs:1871]: If `self.broadcast.is_some()` is true, wouldn't this short circuit and `zk_persist_nonce_update` potentially keep on being `PersistNext` when it should in fact reset?

Comment (nbaztec) [reply crates/cheatcodes/src/inspector.rs:1871]: Good catch!

**Review by elfedy — APPROVED** [review]
LGTM, though @Jrigada should probably review as well

**Review by Karrq — COMMENTED** [review]
Agreed on waiting for @Jrigada

**Comment by nbaztec** [comment]
@Jrigada can you review this please?

**Review by Jrigada — APPROVED** [review]
Thank you Nish

**Pull request merged**